src/frontend/src/pages/GanttPage/GanttChartComponents/GanttTaskBar/GanttTaskBarDisplay.tsx

```diff
@@ -0,0 +1,170 @@
+import { Box, IconButton, Typography, useTheme } from '@mui/material';
+import { grey } from '@mui/material/colors';
+import { ArrowDropDownIcon, ArrowRightIcon } from '@mui/x-date-pickers';
+import { useHistory } from 'react-router-dom';
+import { dateToString } from '../../../../utils/datetime.utils';
+import {
+  GanttTask,
+  isHighlightedChangeOnGanttTask,
+  RequestEventChange,
+  transformWorkPackageToGanttTask
+} from '../../../../utils/gantt.utils';
+import { routes } from '../../../../utils/routes';
+import { wbsPipe } from 'shared';
+import {
+  ganttTaskBarBackgroundStyles,
+  ganttTaskBarContainerStyles,
+  taskNameContainerStyles,
+  webKitBoxContainerStyles,
+  webKitBoxStyles
+} from './GanttTaskBarDisplayStyles';
+import { CSSProperties } from 'react';
+
+interface GanttTaskBarDisplayProps {
+  days: Date[];
+  task: GanttTask;
+  isProject: boolean;
+  handleOnMouseOver: (e: React.MouseEvent, task: GanttTask) => void;
+  handleOnMouseLeave: () => void;
+  onWorkPackageToggle?: () => void;
+  showWorkPackages?: boolean;
+  highlightedChange?: RequestEventChange;
+  getStartCol: (start: Date) => number;
+  getEndCol: (end: Date) => number;
+}
+
+const GanttTaskBarDisplay = ({
```

Comment:
just throwing this comment here, this spacing needs to get fixed:
![image](https://github.com/Northeastern-Electric-Racing/FinishLine/assets/29521172/63de0f1f-3e32-48de-b888-aa4f8917c504)